### DISCUSSION TITLE
[nvim noob]: looking for ready-made nixvim config snippets helping migration from vscode

### DISCUSSION THREAD

**Discussion by ppenguin** [nix-community/nixvim#609] [Q&A] [answered]
I'm a long-term NixOS user and also familiar with vanilla vim but using vscode as my main IDE for multiple languages and also occasionally relying on extensions like ThunderClient (similar to Postman) and DatabaseClient. (I have 68 extensions installed of which I frequently use about 15~20, including `yaml` with `ansible` and `gitlab-ci` schemas, `markdown`, `go`, `ts`, `Makefile`, etc.)

Being intrigued by `neovim` but observing that its plugin manager is yet another package manager bound to collide with `nix` declarative, `nixvim` is IMHO exactly the right abstraction to integrate `neovim` in deployments, especially since it can offer the great bonus of devshell specific IDE configuration.

What I'm really missing now is references to good starter configs in `nixvim`, with sensible/usable generic settings and an explanation/reference where the actual (default) config is specified. For example, I can do `telescope.enable = true` but I have no idea what key-bindings actually do something (per default) or which inter-plugin dependencies need to be defined to make it actually do something useful. Same for LSP support and things like `NvimTree` (which is hardly useful without activation keybindings and/or some automation).

In other words, I'm looking for a realistic "QuickStart" path because atm I can't afford the time to dive deep yet again...

**Comment by GaetanLepage** [comment, marked as answer]
Hi !
Welcome to Nixvim. I indeed think that your use case fits very well into the scope of Nixvim.

To answer your questions:

- The README is a good point to get a grasp of what is possible in Nixvim as well as "how to do the normal things, like keymaps, options, autocmd...". We are working on better documentation but it is not quite there yet.
- Our general philosophy is to do nothing by ourselves except when it makes obvious sense. For instance, if you set `plugins.lsp.servers.pylsp.enable = true`, then we would install the `pylsp` language server (from `nixpkgs`). You still have the liberty to disable that. But apart from this, we take no decisions for you. When you install a plugin, we do not configure it for you.

Nixvim is more the natural layer needed to configure neovim through a nix module than an opinionated neovim distribution.
Also, you can always have a look at the generated `init.lua`. Most of the time, what we are doing is a direct "translation" from nix to lua.

As for configuration examples, you can look at mine for inspiration: https://github.com/GaetanLepage/dotfiles/tree/master/home/modules/tui/neovim

Feel free to join our matrix space to quickly get answers from maintainers and users :)
https://matrix.to/#/#nixvim:matrix.org
